Source/Charts/Renderers/ChartDataRendererBase.swift

```diff
@@ -15,13 +15,13 @@ import CoreGraphics
 @objc(ChartDataRendererBase)
 open class DataRenderer: Renderer
 {
-    @objc open var animator: Animator?
+    @objc open let animator: Animator
```

Comment:
did Xcode tell you to use `let` instead of `var`? I wonder if any chance people would modify renderer's animator later.

Reply:
I made it `let`. What is the use case of changing the renderer?

Reply:
Not sure, just seeing it used to be `var`, though it was optional and some advanced user maybe? Using let will be more strict.
If we use `var`, will Xocde complain it is never re-assigned?

Reply:
It's an instance variable, so it's impossible for Xcode to know if it is never re-assigned.

It is more strict, but from what I can tell there is no consideration for having multiple animators. We have one class for it, and no protocols to define an interface. Overriding any functions in `Animator`, without diving into the framework to know exactly what needs to be there and what can be changed, will cause problems.